build.sbt

```diff
@@ -667,6 +672,61 @@ lazy val test = project
         result.copy(overall = TestResult.Error)
       }
       else result
+    },
+    // The default JUnitXMLListener doesn't play well with partest: we end up clobbering the one-and-only partest.xml
+    // file on group of tests run by `testAll`, and the test names in the XML file don't seem to show the path to the
+    // test for tests defined in a single file.
+    //
+    // Let's roll our own to try to enable the Jenkins JUnit test reports.
+    testListeners in IntegrationTest += new TestsListener {
+      val delegate = new JUnitXmlTestsListener(target.value.getAbsolutePath)
+      import java.util.EnumSet
+      import sbt.testing.{Status => TStatus}
+      val errorStatus = EnumSet.of(TStatus.Error)
+      val failStatus = EnumSet.of(TStatus.Failure)
+      val skipStatus = EnumSet.of(TStatus.Skipped, TStatus.Ignored)
+
+      override def doInit(): Unit = ()
+      override def doComplete(finalResult: TestResult.Value): Unit = ()
+      override def endGroup(name: String, t: Throwable): Unit = ()
+      override def endGroup(name: String, result: TestResult.Value): Unit = ()
+      override def testEvent(event: TestEvent): Unit = {
+        // TODO get the test timings out, too.
+        event.detail.foreach { e =>
+          def count(statusSet: java.util.Set[sbt.testing.Status]) = if (statusSet.contains(e.status())) 1 else 0
+          val trace: String = if (e.throwable.isDefined) {
+            val stringWriter = new StringWriter()
+            val writer = new PrintWriter(stringWriter)
+            e.throwable.get.printStackTrace(writer)
```

Comment:
Should this be processed further (as in https://github.com/sbt/junit-interface/blob/master/src/main/java/com/novocode/junit/RichLogger.java) or does the consumer of the log take care of that?